### PR TITLE
Feat/add kafka out sasl

### PR DESCRIPTION
**Pull request by Tomvbe**

### CODE REVIEW

File: ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioKafkaOut.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Override
+	public void accept(Model model) {
+		kafkaTemplate.send(createProducerRecord(lang, topic, model));
+	}
+
+	private ProducerRecord<String, String> createProducerRecord(Lang lang, String topic, Model model) {
+		final String message = toString(lang, model);
+		final var headers = new RecordHeaders().add(CONTENT_TYPE, lang.getHeaderString().getBytes());
+		return new ProducerRecord<>(topic, null, (String) null, message, headers);
```
Comment (sandervd) [inline ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioKafkaOut.java:33]: The message key should be set to the URI of the state object

File: ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioKafkaOut.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldio;
+
+import be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiOutput;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFWriter;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.springframework.kafka.core.KafkaTemplate;
```
[the remark this answers is not in the document]
Comment (Tomvbe) [reply ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioKafkaOut.java:9]: We use the kafka modules from spring in both kafka in and out as the orchestrator is a spring boot application

File: ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioKafkaOut.java (hunk 1 above)
Comment (Tomvbe) [reply ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioKafkaOut.java:33]: I created a task to implement this in the next PR https://vlaamseoverheid.atlassian.net/browse/VSDSPUB-596

File: ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/test/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioKafkaOutAutoConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		var configurator = new LdioKafkaOutAutoConfig().ldiKafkaOutConfigurator();
+
+		Map<String, String> config = getBasicConfig();
+		config.put(KafkaOutConfigKeys.SECURITY_PROTOCOL, "Fantasy protocol");
+		ComponentProperties componentProperties = new ComponentProperties(config);
+
+		IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+				() -> configurator.configure(componentProperties));
+
+		assertEquals("Invalid 'security-protocol', the supported protocols are: [NO_AUTH, SASL_SSL_PLAIN]",
```
Comment (Yalz) [inline ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/test/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioKafkaOutAutoConfigTest.java:24]: magic string

**Review by Yalz — APPROVED** [review]

**Pull request merged**